### DISCUSSION TITLE
Is it possible for a speaker diarization model to give number of outputs as rows greater than that provided in the input RTTM file?

### DISCUSSION THREAD

**Discussion by ishansharma1320** [NVIDIA-NeMo/NeMo#5209] [Q&A]
To give a context:

I am using speech brain for VAD and am passing the same to the speaker diarization model using oracle_vad true in Speaker Diarization Inference.ipynb.

The output rttm with speaker labels have more number of records than the input rttm.

Is this the intended behaviour, if yes, can someone explain why this happens, and is there any way for both input and output rttms to have same records?

**Comment by tango4j** [comment]
@ishansharma1320 I think the terminologies you are using are not aligned with the terms that NeMo contributors are using. 
Could you give us some examples of "number of records"? 
What is number of outputs in RTTM? Do you mean (1) number of speakers (2) number of lines in RTTM file. It would be great if you can clarify these.

**Reply by ishansharma1320** [reply]
Hi @tango4j

Thanks for replying 

I mean the number of lines in the RTTM file

**Comment by tango4j** [comment]
- Speaker diarization is also a machine learning task where outputs are estimated results. Therefore it does not guarantee 100% correct results in any case.
- In this sense, output RTTM lines are containing the start and end timestamps for a segment. This could be different from the ground-truth RTTM (I am not sure what "input RTTM" means but if I assume it is reference RTTM)
- The output RTTM will very rarely have the same number of lines with the given groun-truth RTTM. This happens because speaker diarization is an estimation process based on a model.

**Reply by ishansharma1320** [reply]
Hi @tango4j,

Thanks for your reply,

That clears it. Thanks
